common/src/main/java/com/revenuecat/purchases/common/OfferingFactories.kt

```diff
@@ -0,0 +1,66 @@
+package com.revenuecat.purchases.common
+
+import com.revenuecat.purchases.Offering
+import com.revenuecat.purchases.Offerings
+import com.revenuecat.purchases.Package
+import com.revenuecat.purchases.PackageType
+import com.revenuecat.purchases.models.StoreProduct
+import com.revenuecat.purchases.strings.OfferingStrings
+import org.json.JSONObject
+
+/**
+ * Note: this may return an empty Offerings.
+ */
+fun JSONObject.createOfferings(productsById: Map<String, StoreProduct>): Offerings {
+    val jsonOfferings = getJSONArray("offerings")
+    val currentOfferingID = getString("current_offering_id")
+
+    val offerings = mutableMapOf<String, Offering>()
+    for (i in 0 until jsonOfferings.length()) {
+        val offeringJson = jsonOfferings.getJSONObject(i)
+        offeringJson.createOffering(productsById)?.let {
+            offerings[it.identifier] = it
+
+            if (it.availablePackages.isEmpty()) {
+                warnLog(OfferingStrings.OFFERING_EMPTY.format(it.identifier))
+            }
+        }
+    }
+
+    return Offerings(offerings[currentOfferingID], offerings)
+}
+
+fun JSONObject.createOffering(productsById: Map<String, StoreProduct>): Offering? {
```

Comment:
only change besides pulling into a new file is renaming the parameters from `products` -> `productsById` and pulling out some intermediate variables in the json parsing